File: script/disease_phenotype_ordo.py
from __future__ import annotations

from pathlib import Path

from disease_phenotype_association_util import (
    HPOA_SOURCE_URI,
    HPO_PHENOTYPE_PATH,
    ORPHANET_PRODUCT4_PATH,
    RDF_DIR,
    create_annotation_source,
    load_manual_phenotype_associations,
    load_ordo_frequency_annotations,
    write_ordo_phenotype_association_ttl,
)


def main() -> None:
    orphanet_frequency = load_ordo_frequency_annotations(ORPHANET_PRODUCT4_PATH)
    print(f"Orphanet_frequency Count : {len(orphanet_frequency)}")

    orphanet_manual = load_manual_phenotype_associations(HPO_PHENOTYPE_PATH, "ORPHA")
    print(f"Orphanet_HPO_Manual Count : {len(orphanet_manual)}")

    source = create_annotation_source("Orphanet", HPOA_SOURCE_URI)

    write_ordo_phenotype_association_ttl(
        Path(RDF_DIR) / "Orphanet_HP_Association.ttl",
        orphanet_manual,
        orphanet_frequency,
        source,
    )
    print(f"Orphanet_HPO_Association Count : {len(orphanet_manual)}")


if __name__ == "__main__":
    main()

File: scripts/java/DiseasePhenotypeORDO.java
package pubcasefinder_260415;

import java.util.LinkedHashMap;

// Orphanet-HPO 표현형 연관과 빈도 주석을 결합해 Orphanet_HP_Association.ttl을 생성한다.
public class DiseasePhenotypeORDO {
	/** Orphanet 빈도 정보와 수동 HPO 연관을 합쳐 Orphanet_HP_Association.ttl을 생성한다. */
	public static void main(String[] args) throws Exception {
		LinkedHashMap<String, String> orphanetFrequency =
				DiseasePhenotypeAssociationUtil.loadOrdoFrequencyAnnotations(
						DiseasePhenotypeAssociationUtil.ORPHANET_PRODUCT4_PATH);
		System.out.println("Orphanet_frequency Count : " + orphanetFrequency.size());

		LinkedHashMap<String, String> orphanetManual =
				DiseasePhenotypeAssociationUtil.loadManualPhenotypeAssociations(
						DiseasePhenotypeAssociationUtil.HPO_PHENOTYPE_PATH,
						"ORPHA");
		System.out.println("Orphanet_HPO_Manual Count : " + orphanetManual.size());

		DiseasePhenotypeAssociationUtil.AnnotationSource source =
				DiseasePhenotypeAssociationUtil.createAnnotationSource(
						"Orphanet",
						DiseasePhenotypeAssociationUtil.HPOA_SOURCE_URI);

		DiseasePhenotypeAssociationUtil.writeOrdoPhenotypeAssociationTtl(
				DiseasePhenotypeAssociationUtil.RDF_DIR + "/Orphanet_HP_Association.ttl",
				orphanetManual,
				orphanetFrequency,
				source);
		System.out.println("Orphanet_HPO_Association Count : " + orphanetManual.size());
	}
}
